### PR TITLE
Adds major warning text for some jobs

### PR DESCRIPTION
**Pull request by AffectedArc07**
## What Does This PR Do
Adds major warnings to jobs regarding crucial things that people should see even if they dont read the wiki

image

## Why It's Good For The Game
This was born from people in the Discord saying there are too many IAAs who aren't playing the job properly at all, and they are treating it as if they are the governor of SoP (you aren't). This PR adds a very obvious warning to those taking the role, hopefully alleviating these issues.

## Changelog
:cl: AffectedArc07
add: Added very obvious warnings for how to play some roles which can make or break a round for people
/:cl:

### CODE REVIEW

File: code/game/jobs/job/supervisor.dm
Comment (AffectedArc07) [inline code/game/jobs/job/supervisor.dm:259]: Source:
image
image

**Comment by DoctorDrugs** [comment]
The IAA one is fine in my opinion, but i have issues with the security officer one as I've noticed for a long time now that new officers and even some of the vets will toss people into cells for the maximum allowed time for a sentence just because it's the max. Rather than gauging if it's someone who deserves the max time they just simply give it cause of space law, there's probably a better way to word the one for security officers but it's not within reach of my early morning brain.

**Comment by AffectedArc07** [comment]
> The IAA one is fine in my opinion, but i have issues with the security officer one as I've noticed for a long time now that new officers and even some of the vets will toss people into cells for the maximum allowed time for a sentence just because it's the max. Rather than gauging if it's someone who deserves the max time they just simply give it cause of space law, there's probably a better way to word the one for security officers but it's not within reach of my early morning brain.

If you (or anyone else for that matter) thinks of something better, I'll gladly change it.

**Comment by Miraviel** [comment]
If this is approved, you could just add it to #17419 since in its current form, it'll throw off #17419's formatting (this red text message will be displayed below the ------- line if mine gets merged, aligned to the left, while my box is centered).

Or wait until it gets merged and rebase it on top of it?

Or I can rewrite #17419's logic to better fit yours?

Not sure which one you prefer but I'm down for all of these

**Comment by AffectedArc07** [comment]
> If this is approved, you could just add it to #17419 since in its current form, it'll throw off #17419's formatting (this red text message will be displayed below the ------- line if mine gets merged, aligned to the left, while my box is centered).
> 
> Or wait until it gets merged and rebase it on top of it?
> 
> Or I can rewrite #17419's logic to better fit yours?
> 
> Not sure which one you prefer but I'm down for all of these

I am happy to wait for your PR to go in to ease merge conflict resolution

**Comment by Sirryan2002** [comment]
Not a negative change, I just think that given that all of this information Is made abundantly clear in the linked wiki pages already this may just be unnecessary.

**Comment by lewcc** [comment]
At this point, why not give the detective a glaring red message that says YOUR REVOLVER IS FOR SELF DEFENSE

**Comment by Carthusia** [comment]
Maybe for magistrate it could say somthing like: (Security SOP violations should be accounted for during sentencing during non-emergencies)
Making it very clear to magistrates that yes, you CAN throw out a case entirely thats not somthing like murder or blatent EOC activity, because security completely ignored their SOP on green/blue

Perhaps for the security officer one, also have it note: (Failure to follow your SOP during non-emergencies may severely affect a suspects chance of being convicted)

Implying that by failing to follow your SOP during an arrest on lower alert levels and outside of emergencies, the entire case may just be thrown out by a magistrate or the HOS.

Ive seen many a person on green taze/stunned, cuffed, and dragged off into the brig, completely ignoreing SOP, it was FAR worse back when we had security borgs who by design didnt need to follow SOP and would just taze/stun/cuff/drag every time because "Im not security, im a borg"

**Comment by AffectedArc07** [comment]
> Not a negative change, I just think that given that all of this information Is made clear in the linked wiki pages already this may be unnecessary.

If everyone read the wiki, we wouldn't need this. However, people think they can go into a role they have never played before and act like they know how to do everything, and that leads to tons of complaints from numerous people.

**Comment by AffectedArc07** [comment]
> At this point, why not give the detective a glaring red message that says YOUR REVOLVER IS FOR SELF DEFENSE

Unironically this. Something like `You are not a Security Officer. Your revolver is for self defence only.`

**Comment by MarcellusPye** [comment]
Synthetics could probably use a warning about their laws and following the AI's orders. Anything else I can think of has a playtime requirement, at which point this shouldn't be necessary. I'll throw in a few notes I made though.

Heads: Something about a higher knowledge of SoP and seriousness being required
     "This role requires you to coordinate your department. You are required to be familiar with SoP, and act with reasonable seriousness"
Blueshield: Emphasising that they're there to protect the heads, anything else should be directed to security
    "You are a bodyguard. You should only apprehend someone if they have directly impacted/attacked a member of Command or a Centcom VIP"
NT rep: Possibly a warning about faxes, probably isn't necessary
Magistrate: A warning about how serious the role is
   "Magistrates are the absolute authority on space law. They must follow SoP/Space Law strictly, and act seriously."

**Comment by Shadeykins** [comment]
> Heads: Something about a higher knowledge of SoP and seriousness being required "This role requires you to coordinate your department. You are required to be familiar with SoP, and act with reasonable seriousness"

"This role requires you to coordinate a large department. You are required to be familiar with Standard Operating Procedure (Department), basic job duties, and must act in a professional and reasonable manner (roleplay)."

>Blueshield: Emphasising that they're there to protect the heads, anything else should be directed to security "You are a bodyguard. You should only apprehend someone if they have directly impacted/attacked a member of Command or a Centcom VIP"

"This role requires you to ensure the safety of the Heads of Staff, not the general crew. You may perform arrests only if the combatant is directly threatening a member of Command, the Nanotrasen Representative, or the Magistrate."

> NT rep: Possibly a warning about faxes, probably isn't necessary

"This role requires you to advise the Command team about Standard Operating Procedure, Chain of Command, and report to Central Command about various matters. You are required to act in a manner befitting someone representing Nanotrasen."

> Magistrate: A warning about how serious the role is "Magistrates are the absolute authority on space law. They must follow SoP/Space Law strictly, and act seriously."

"This role requires you to oversee legal matters and make important decisions about sentencing. You are required to have an extensive knowledge of Space Law and Security SOP and only operate within, not outside, the boundaries of the law."

**Comment by MarcellusPye** [comment]
Thanks, I'm not the best at wording things.

File: code/game/jobs/job/engineering.dm
Comment (AffectedArc07) [inline code/game/jobs/job/engineering.dm:24]: @Shadeykins Did you mean for me to replace `(Department)` with that department's SOP?

File: code/game/jobs/job/security.dm
Comment (AffectedArc07) [inline code/game/jobs/job/security.dm:25]: Looking at this compared to other ones in the file, I feel like these should be shorter and to the point.

File: code/game/jobs/job/engineering.dm
Comment (Shadeykins) [reply code/game/jobs/job/engineering.dm:24]: yes

**Comment by Sirryan2002** [comment]
> > Not a negative change, I just think that given that all of this information Is made clear in the linked wiki pages already this may be unnecessary.
> 
> If everyone read the wiki, we wouldn't need this. However, people think they can go into a role they have never played before and act like they know how to do everything, and that leads to tons of complaints from numerous people. 

Fair point

File: code/game/jobs/job/security.dm
Comment (Shadeykins) [reply code/game/jobs/job/security.dm:25]: "This role requires you to coordinate a department. You are required to be familiar with Standard Operating Procedure (Department), basic job duties, and act professionally (roleplay)."

**Comment by Tenheydes** [comment]
OK, so...  I really like this commit, but I'm going to step back and ask... why are we doing this, so late in the process when they're already in round, and can't change their choice, and are already in the map where they can't safely tab out to a browser and start learning space law while being assaulted in the background window?

Why is this not a popup when a player clicks "enter game" and clicks a crewmember type.

Keep a bitmask of jobs that are round-spoiling, and if a player has read.  

The first time a new player clicks security officer in the round join screen, it pops this same kind of intro text in a dialog box.  "No seriously this is a hard job, you need to go read and at least have a basic understanding of space law and Sec SOP, before you enter the round as a sec player.<br><br>Failure to play this role according to in-character SOP may result in being temporarily or permanently banned from playing this role."  and the buttons at the bottom are, "agree" and "return" -- And the first time they click "agree" it sets the "UNDERSTANDS_SEC" flag and the text is never shown again and then they enter the game.  Once the "UNDERSTANDS_SEC" is set, they enter the game with no popup.

If it's the character editor "job preferences" screen, then if the UNDERSTANDS_SEC flag is not set, as soon as you click from "NONE" to "LOW" then the box pops up at that moment, and if they hit accept it sets the flag and lets them turn on the job, if they click "return" then it sets the job back to "NONE"

All the important jobs are locked to time as crew, right?  So even with the dialog boxes, there's low (or at least less?) risk of brand new player "yeah whatever, I just want to play, hit ok" ignore-the-box style behavior

**Review by dearmochi — APPROVED** [review]

**Pull request merged**
